src/frontend/apps/standalone_site/src/features/Playlist/features/UpdatePlaylist/components/AddUserAccessForm.tsx

```diff
@@ -0,0 +1,165 @@
+import { ReactComponent as SearchPeopleIcon } from 'assets/svg/iko_search-people.svg';
+
+import { useCreatePlaylistAccess } from '../api/useCreatePlaylistAccess';
+import { UserLite } from '../api/useSearchUsers';
+import { PlaylistRole } from '../types/playlistAccess';
+
+import { SearchUserListRow } from './SeachUserListRow';
+import { SearchUserList } from './SearchUserList';
+import { userRoleOptions } from './UserRoleOptions';
+
+const messages = defineMessages({
+  title: {
+    defaultMessage: 'Add member',
+    description: 'Add member to playlist modal title.',
+    id: 'features.Playlist.features.UpdatePlaylist.components.AddUserAccessForm.title',
+  },
+  back: {
+    defaultMessage: 'back',
+    description: 'Back to user selection button title.',
+    id: 'features.Playlist.features.UpdatePlaylist.components.AddUserAccessForm.back',
+  },
+  roleSelectLabel: {
+    defaultMessage: 'Select role',
+    description: 'Select role for user to add in playlist.',
+    id: 'features.Playlist.features.UpdatePlaylist.components.AddUserAccessForm.roleSelectLabel',
+  },
+  addMember: {
+    defaultMessage: 'Add this member',
+    description: 'Add member to playlist button title.',
+    id: 'features.Playlist.features.UpdatePlaylist.components.AddUserAccessForm.addMember',
+  },
+  searchInfos: {
+    defaultMessage:
+      'Search within user and select the one you want to add to this playlist.',
+    description: 'Modale description to search user to add into playlist.',
+    id: 'features.Playlist.features.UpdatePlaylist.components.AddUserAccessForm.searchInfos',
+  },
+  placeHolder: {
+    defaultMessage: 'Search by username or email address...',
+    description: 'Search user input placeholder.',
+    id: 'features.Playlist.features.UpdatePlaylist.components.AddUserAccessForm.placeHolder',
+  },
+});
+
+interface AddUserAccessFormProps {
+  playlistId: string;
+  onUserAdded: () => void;
+  excludedUsers?: string[];
+}
+
+export const AddUserAccessForm = ({
+  playlistId,
+  onUserAdded,
+  excludedUsers,
+}: AddUserAccessFormProps) => {
+  const intl = useIntl();
+
+  const options = useMemo(() => userRoleOptions(intl), [intl]);
+  const [typedStr, setTypedStr] = useState('');
+  const [searchedUser, setSearchedUser] = useState('');
+  const [selectedUser, setSelectedUser] = useState<Nullable<UserLite>>(null);
+  const [roleValue, setRoleValue] = useState(
+    options.find((option) => option.key === PlaylistRole.STUDENT),
+  );
+  const { mutate: createPlaylistAccess } = useCreatePlaylistAccess({
+    onSuccess: () => {
+      onUserAdded();
+    },
+  });
+
+  const debouncedSetSearchedStr = useRef(
+    debounce((str: string) => setSearchedUser(str), 1000),
+  ).current;
+
+  // Prevent debounce leak
+  useEffect(() => {
+    return () => {
+      debouncedSetSearchedStr.cancel();
+    };
+  }, [debouncedSetSearchedStr]);
+
+  return (
+    <Box>
+      <Heading alignSelf="center">{intl.formatMessage(messages.title)}</Heading>
```

Comment:
We should try to keep the same heading between modals.

https://github.com/openfun/marsha/blob/feature/vassili/create-playlist-accesses/src/frontend/apps/standalone_site/src/features/Contents/features/ClassRoom/components/Create/ClassRoomCreate.tsx#L65